utility/doc/E2E_FEATURE_PATH_TEMPLATE.md

```diff
@@ -0,0 +1,218 @@
+# E2E Feature Path <!-- omit in toc -->
+
+_IMPROVE(olshansky): Once we've completed the entire process at least once, we'll add links to each step._
+
+- [Introduction \& Goals](#introduction--goals)
+- [Developer Journey](#developer-journey)
+- [Feature Specification](#feature-specification)
+  - [Spot E2E Feature](#spot-e2e-feature)
+  - [Spike E2E Feature](#spike-e2e-feature)
+  - [Scope E2E Feature](#scope-e2e-feature)
+    - [GitHub Ticket](#github-ticket)
+    - [Origin Document](#origin-document)
+- [E2E Feature Implementation](#e2e-feature-implementation)
+  - [POC: Proof of Concept](#poc-proof-of-concept)
+  - [MVP: Minimum Viable Product](#mvp-minimum-viable-product)
+  - [PROD: Production](#prod-production)
+
+## Introduction & Goals
+
+The [Pocket Network Specification](https://github.com/pokt-network/pocket-network-protocol/tree/main/utility) implementation is driven by various [milestones](https://github.com/pokt-network/pocket/milestones) and protocol/module/component specific tasks. Each feature cross the boundaries of business logic, data types, and interfaces for different components. Due to the complex nature of implementation, we've designed a more streamlined approach.
+
+**The goal** of this document is to outline a well-defined process for adopting an end-to-end feature path approach. This makes each feature/task easier to scope, reason about, design, and implement.
+
+## Developer Journey
+
+```mermaid
+%%{init: { 'logLevel': 'debug', 'theme': 'base' } }%%
+  timeline
+    title E2E Feature Path Developer Journey
+
+    section E2E Feature Specification
+      Spot Feature:
+        User research:
+        Product management:
+        Protocol intuition/experience:
+        Addition/removal/selection of feature from the list
+      Spike Feature:
+        Research details:
+        Identify pointers:
+        Note dependencies:
+        Find blockers
+      Scope Feature:
+        Define requirements:
+        Document E2E implementation
+
+    section E2E Feature Implementation
+        POC:
+          POC Spike:
+          Explore:
+          Hack:
+          Have fun!
+        MVP:
+          Data Structures:
+          Interfaces:
+          Implementation:
+          Unit Tests:
+          E2E Tests:
+          Documentation
+        PROD:
+          Identify workarounds & hacks:
+          Document future work:
+          Ideate!
+```
+
+## Feature Specification
+
+### Spot E2E Feature
+
+Choose or add a feature from the Utility E2E feature list [here](./E2E_FEATURE_LIST.md).
```

Comment:
Is it intentional that this file isn't present yet?

Reply:
Yea, that is up for review in #766 and should be merged in shortly as well.